src/pages/AdvancedSearch/SearchForm/SearchHistoryMenu/SearchHistoryMenu.tsx

```diff
@@ -0,0 +1,54 @@
+import { ContextualMenu, Icon } from "@canonical/react-components";
+import { useFormikContext } from "formik";
+
+import { useQueryParams } from "hooks/useQueryParams";
+
+import type { FormFields } from "../types";
+
+export enum Label {
+  CLEAR = "Clear history",
+  HISTORY = "History",
+}
+
+type Props = {
+  queryHistory: string[];
+  setQueryHistory: (queryHistory: string[]) => void;
+};
+
+const SearchHistoryMenu = ({
+  queryHistory,
+  setQueryHistory,
+}: Props): JSX.Element => {
+  const { setFieldValue } = useFormikContext<FormFields>();
+  const [, setQueryParams] = useQueryParams<{ q: string }>({
+    q: "",
+  });
+
+  return (
+    <ContextualMenu
+      links={[
+        ...queryHistory.map((query) => ({
+          children: query,
+          onClick: () => {
+            setQueryParams({ q: encodeURIComponent(query) });
+            setFieldValue("query", query);
+          },
+        })),
+        {
+          children: (
+            <>
+              <Icon name="delete" /> {Label.CLEAR}
+            </>
+          ),
+          onClick: () => setQueryHistory([]),
+        },
+      ]}
+      position="left"
+      toggleClassName="has-icon"
+      toggleDisabled={queryHistory.length === 0}
```

Comment:
I was wondering if maybe we could also disable it when `isCrossModelQueryLoading = true`. Could we have 2 requests being done almost instantly (by pressing 2 values from history one after another) and then having the first `updateCrossModelQuery` finish after the second one, thus getting the wrong result?

Reply:
Yep, good thinking. I've added that.